lib/scion.js

```diff
@@ -815,10 +815,26 @@
               return actionRef.call(this._scriptingContext, currentEvent);     //SCXML system variables
             } catch (e){
               var err = {
-                tagname: actionRef.tagname, 
-                line: actionRef.line, 
-                column: actionRef.column,
-                reason: e.message
+                reason: e.message,
+                event: currentEvent
+              };
+              if (typeof actionRef == 'function') {
+                var fnStr = actionRef.toString();
+                var m = fnStr.match(/line_([0-9]+)_column_([0-9]+)\(/);
+                if (m) {
+                  // SCXML
+                  err.line = parseInt(m[1]);
+                  err.column = parseInt(m[2]);
+                } else {
+                  // default
+                  err.actionRef = fnStr;
+                }
+              } else {
+                // I don't know when this can happen since actionRef.call() is always called before
+                // Kept it for backward compatibility but could/should remove it
+                err.tagname = actionRef.tagname;
```

Comment:
I perfectly agree with you that this is hacky. 
Maybe we can figure out something cleaner.

Here is the basic scenario I'm trying to solve :
```javascript
var scxml = require('scxml');
var scion = require('scion-core');

scxml.pathToModel('test.xml', function(err, model) {
	if (err) throw err;
	// use scion-core last version (git+https://github.com/jbeard4/SCION-CORE.git)
	// because scxml.scion does not implement the onError handler
	var statechart = new scion.Statechart(model);
	statechart.registerListener({
		onError: function(err) {
			console.log(err);
  	    }
	});
	statechart.start();
});
```
with text.xml being:
```xml
<scxml xmlns="http://www.w3.org/2005/07/scxml" version="1.0" initial="start">
  <state id="start">
    <onentry>
      <!-- expr is a javascript error -->
      <log label="test" expr="m.test()"/>
    </onentry>
  </state>
</scxml>
```
And this gives:
```
{ tagname: undefined,
  line: undefined,
  column: undefined,
  reason: 'm is not defined' }
```
How do you propose to solve this (properly) ?

Cheers,
Vincent